software-engineering-policies/Containers/ContainerPolicy.md

```diff
@@ -1,55 +1,23 @@
-* Monitoring solution - ELK (.Net) EFK (Java) stacks, Graphana, Prome...(learn to spell)
-* Utilise automated deployments on version increments within the container registry out to the nominated environment (eg dev)
+Containers are in use in Data Science and Software Engineering, in all environments of development, test and production. This policy defines the use and deployment for current and future usage.
 
-### Could do
+## Policy
 
-* Utilise DAPR for sidecar third party component management (queues, DB's etc)
-* Utilise native built in support within visual studio for Docker
-* Utilise Docker Compose to stub other dependencies that container utilises
-
-## Best practice guides
-
-<https://sysdig.com/blog/dockerfile-best-practices/>
-
-Some notes from a previous principal dev on Docker/containers
-
-1. Skills - we should look to keep stacks quite similar amongst the different teams within Software Engineering (without stifling innovation), to facilitate rotation but also the transfer of services from team to support team. I would not want a delivery to be using containers and the destination support team to have no skills in them. This could arguably be covered as part of the handover and up front support design of a solution, but may differ from team to team. Do we need formal training in containers across SE? Or probably at the very least guidance on the use of, to prevent teams achieving the same thing in 100 different ways?
+* Containers are a potential option for architects to utilise (a tool in the toolbox).
+* Consideration should be made around why we want to utilise containers.
+* Consideration should be made around who will inherit the environments at the outset. i.e. is there expertise in how manage the hosting environment?
+* Utilisation of Containers in development and test environments is acceptable without APF review.
+* In the Software Engineering department, deployment of containers into live will need to be approved by The Head of Software Engineering in conjunction with the Principal and Lead Software Engineers, until agreement is reached via APF forum.
+* Containers are a core software and test engineering tool.
+* [Container best practices](./ContainerBestPractices.md) should be adopted by all teams using containers.
+* The Lead Software Engineer for each team is responsible for ensuring that this policy is adhered.
 
-1. Security - should we have a centralised container registry? As containers work from images pulled from the internet (albeit a supposedly "trusted" source such as Docker hub), do we need more rigour around that? When we spoke about containers a while back - when we had the ALM team - the vision was that they would own a central container registry that teams would pull from. If a team wanted a new container, this would firstly be added to our registry and the team would pull from there. This would enable a sheep-dip of container images, if we thought this was necessary (ITSO should have a say here).
```

Comment:
Has the decision been made not to have a centrally managed container registry?

Reply:
This policy can be updated if we take the decision to have one. In the meantime we are using Dockerhub.